### PR TITLE
Implement file's FileSystem API in node_io

### LINKED ISSUE #86: node_io: Implement `FileSystem` from package:file
Related to https://github.com/dart-lang/tools/issues/1991.

Today the `glob` package uses `node_io` with a conditional export but we want to get away from that. We are switching to use the `file` package for all file system interactions.

If this package had an implementation of `FileSystem`, then it could easily plug in to `glob` in a clean way. This might also help some of the other issues listed for this package, if we can push more of the ecosystem towards using these general file interfaces instead of those from `dart:io`, then everything can be more flexible :).

### PR DESCRIPTION
**Pull request by nex3**
Closes #86

### CODE REVIEW

File: node_io/pubspec.yaml [hunk 1]
```diff
@@ -18,3 +19,6 @@ dev_dependencies:
   build_runner: ">=1.10.0 <1.10.2"
   build_node_compilers: ^0.3.0
   build_test: ^1.0.0
+
+dependency_overrides:
+  node_interop: {path: ../node_interop}
```
Comment (pulyaevskiy) [inline node_io/pubspec.yaml]: This is no longer needed, right?

**Review by pulyaevskiy — CHANGES_REQUESTED** [review]
LGTM, thanks! Just one minor comment in pubspec.

File: node_io/pubspec.yaml (hunk 1 above)
Comment (nex3) [reply node_io/pubspec.yaml]: Done.

**Review by pulyaevskiy — APPROVED** [review]
LGTM, thanks again!

**Pull request merged**